torch/csrc/jit/script/module.cpp

```diff
@@ -368,6 +369,37 @@ void Module::dump(
             << std::endl;
 }
 
+namespace {
+void export_opnames(const script::Module& m, std::unordered_set<std::string>& opnames) {
+  auto methods = m.get_methods();
+  for (const auto& method : methods) {
+    const auto& func = method.function();
+    torch::jit::Code code(func.graph());
+    for (size_t i = 0; i < code.instructions().size(); ++i) {
+      auto ins = code.instructions()[i];
+      auto node = code.instructions_source()[i];
+      if (ins.op == OpCode::OP) {
+        auto opname = node->schema().operator_name();
+        std::string namestr = opname.name;
+        if (!opname.overload_name.empty())
+          namestr += "." + opname.overload_name;
+        opnames.emplace(namestr);
+      }
+    }
+  }
+  for (const auto& sub_m : m.children()) {
+    export_opnames(sub_m, opnames);
+  }
+}
+} // namespace
+
+// Returns a list of names of all operators in the module and its submodules.
+std::vector<std::string> Module::opnames() const {
+  std::unordered_set<std::string> names;
+  export_opnames(*this, names);
+  return std::vector<std::string>(names.begin(), names.end());
```

Comment:
Order of elements in the returned vector is non-deterministic since they come from `std::unordered_set`. That might result in hard to debug issues, so please make sure that the order is well-specified.